### PR TITLE
MINIFICPP-1256 - Apply fixes recommended by modernize-equals-default clang tidy check

### PR DESCRIPTION
**Pull request by hunyadi-dev**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/include/core/state/FlowIdentifier.h [hunk 1]
```diff
@@ -68,12 +68,7 @@ class FlowIdentifier {
     bucket_id_ = other.bucket_id_;
     flow_id_ = other.flow_id_;
   }
-  FlowIdentifier &operator=(const FlowIdentifier &other) {
-    registry_url_ = other.registry_url_;
-    bucket_id_ = other.bucket_id_;
-    flow_id_ = other.flow_id_;
-    return *this;
-  }
+  FlowIdentifier &operator=(const FlowIdentifier &other) = default;
```
Comment (szaszm) [inline libminifi/include/core/state/FlowIdentifier.h:71]: Please `=default` the "copy ctor" above as well and remove its `explicit` specifier.

File: libminifi/include/core/state/UpdatePolicy.h [hunk 2]
```diff
@@ -69,9 +69,7 @@ class UpdatePolicy {
   }
 
  protected:
-  explicit UpdatePolicy(const UpdatePolicy &other)
-      : enable_all_(other.enable_all_), properties_(other.properties_) {
-  }
+  explicit UpdatePolicy(const UpdatePolicy &other) = default;
```
Comment (szaszm) [inline libminifi/include/core/state/UpdatePolicy.h:72]: Would you mind fixing the copy and move ctors by removing the `explicit` specifiers and making the move ctor and assignment operators take mutable rvalue reference? Not strictly related to your changes, but a trivial fix nearby.

File: libminifi/include/core/state/UpdatePolicy.h [hunk 3]
```diff
@@ -122,10 +120,7 @@ class UpdatePolicyBuilder {
 
 
  protected:
-  explicit UpdatePolicyBuilder(const UpdatePolicyBuilder &other)
-      : current_policy_(other.current_policy_) {
-  }
-
+  explicit UpdatePolicyBuilder(const UpdatePolicyBuilder &other) = default;
```
Comment (szaszm) [inline libminifi/include/core/state/UpdatePolicy.h:123]: Would you mind fixing this copy ctor by removing the `explicit` specifier?

File: libminifi/include/core/Relationship.h [hunk 4]
```diff
@@ -78,11 +74,7 @@ class Relationship {
     return name_ < right.name_;
   }
 
-  Relationship &operator=(const Relationship &other) {
-    name_ = other.name_;
-    description_ = other.description_;
-    return *this;
-  }
+  Relationship &operator=(const Relationship &other) = default;
```
Comment (szaszm) [inline libminifi/include/core/Relationship.h:77]: :+1:

Not only did you simplify the code, but you also handled self-assignment of many classes, including this one. Thanks.

File: extensions/jni/jvm/JniProcessContext.h [hunk 5]
```diff
@@ -39,8 +39,7 @@ struct JniProcessContext {
       : nifi_processor_(nullptr),
         cslookup_(nullptr),
         processor_(nullptr),
-        context_(nullptr) {
-  }
+        context_(nullptr) = default;
```
[the remark this answers is not in the document]
Comment (hunyadi-dev) [reply extensions/jni/jvm/JniProcessContext.h]: Corrected.

File: extensions/jni/jvm/JavaClass.h [hunk 6]
```diff
@@ -46,9 +46,7 @@ class JavaClass {
  public:
 
   JavaClass()
-      : class_ref_(nullptr) {
-
-  }
+      : class_ref_(nullptr) = default;
```
[the remark this answers is not in the document]
Comment (hunyadi-dev) [reply extensions/jni/jvm/JavaClass.h]: Corrected.

File: extensions/openwsman/processors/SourceInitiatedSubscriptionListener.cpp [hunk 7]
```diff
@@ -159,8 +159,7 @@ SourceInitiatedSubscriptionListener::Handler::Handler(SourceInitiatedSubscriptio
 
 SourceInitiatedSubscriptionListener::SubscriberData::SubscriberData()
     : bookmark_(nullptr)
-    , subscription_(nullptr) {
-}
+    , subscription_(nullptr) = default;
```
[the remark this answers is not in the document]
Comment (hunyadi-dev) [reply extensions/openwsman/processors/SourceInitiatedSubscriptionListener.cpp]: Removed.

File: libminifi/include/core/state/FlowIdentifier.h (hunk 1 above)
Comment (hunyadi-dev) [reply libminifi/include/core/state/FlowIdentifier.h:67]: As requested.

File: extensions/windows-event-log/wel/WindowsEventLog.h [hunk 8]
```diff
@@ -63,8 +63,7 @@ typedef std::vector<std::pair<METADATA, std::string>> METADATA_NAMES;
 class WindowsEventLogHandler
 {
 public:
-  WindowsEventLogHandler() : metadata_provider_(nullptr){
-  }
+  WindowsEventLogHandler() : metadata_provider_(nullptr) = default;
```
[the remark this answers is not in the document]
Comment (hunyadi-dev) [reply extensions/windows-event-log/wel/WindowsEventLog.h]: Corrected.

File: libminifi/include/core/state/UpdatePolicy.h (hunk 3 above)
Comment (hunyadi-dev) [reply libminifi/include/core/state/UpdatePolicy.h]: Done.

File: libminifi/include/core/state/UpdatePolicy.h (hunk 2 above)
Comment (hunyadi-dev) [reply libminifi/include/core/state/UpdatePolicy.h]: Done.

File: extensions/opencv/FrameIO.h [hunk 9]
```diff
@@ -30,7 +30,7 @@ class FrameWriteCallback : public OutputStreamCallback {
     // TODO - Nghia: Check std::move(img_mat).
         : image_mat_(std::move(image_mat)), image_encoding_(image_encoding_) {
     }
-    ~FrameWriteCallback() override = default;
+    ~FrameWriteCallback() = default;
```
[the remark this answers is not in the document]
Comment (hunyadi-dev) [reply extensions/opencv/FrameIO.h]: Restored.

File: libminifi/include/core/state/UpdatePolicy.h [hunk 10]
```diff
@@ -69,11 +69,9 @@ class UpdatePolicy {
   }
 
  protected:
-  explicit UpdatePolicy(const UpdatePolicy &other)
-      : enable_all_(other.enable_all_), properties_(other.properties_) {
-  }
+  UpdatePolicy(const UpdatePolicy &other) = default;
 
-  explicit UpdatePolicy(const UpdatePolicy &&other)
+  UpdatePolicy(UpdatePolicy &&other)
```
Comment (szaszm) [inline libminifi/include/core/state/UpdatePolicy.h:74]: The move ctor could be defaulted, too.

**Comment by arpadboda** [comment]
LGTM, happy to merge after the remaining comments of @szaszm are addressed.

File: libminifi/include/core/state/UpdatePolicy.h (hunk 10 above)
Comment (hunyadi-dev) [reply libminifi/include/core/state/UpdatePolicy.h]: Changed the move constructor and the move assignment as well.

**Pull request closed without merging**